### ISSUE TITLE
File from public share "allow upload and editing" uploaded with error and first shown after refresh website

### ISSUE THREAD

**Issue by devnull4242** [nextcloud/server#23191]
Nextcloud 20, self hosted but also https://try.nextcloud.com ( https://demo1.nextcloud.com and https://demo2.nextcloud.com)

If i use a public share with read- and write access "Allow upload and editing" i can upload files. But first i get an error "Unbekannter Fehler aufgetreten". After refresh i can see the files.

Logs from my server:

```
[PHP] Error: Error: Cannot modify header information - headers already sent by (output started at /var/www/virtual/XXX/html/cloud/3rdparty/sabre/http/lib/Sapi.php:132) at /var/www/virtual/XXX/html/cloud/apps/dav/lib/Connector/Sabre/File.php#691 at <<closure>>

 0. <<closure>>
    OC\Log\ErrorHandler::onError(2, "Cannot modify h ... )", "/var/www/virtua ... p", 691, {string: "X-Hash ... "})
 1. /var/www/virtual/XXX/html/cloud/apps/dav/lib/Connector/Sabre/File.php line 691
    header("X-Hash-MD5: d2d ... d")
 2. /var/www/virtual/XXX/html/cloud/apps/dav/lib/Connector/Sabre/File.php line 189
    OCA\DAV\Connector\Sabre\File->header("X-Hash-MD5: d2d ... d")
 3. <<closure>>
    OCA\DAV\Connector\Sabre\File->OCA\DAV\Connector\Sabre\{closure}("*** sensitive parameters replaced ***")
 4. /var/www/virtual/XXX/html/cloud/lib/private/Files/Stream/HashWrapper.php line 71
    call_user_func(Closure {}, "*** sensitive parameter replaced ***")
 5. <<closure>>
    OC\Files\Stream\HashWrapper->stream_close()
 6. /var/www/virtual/XXX/html/cloud/apps/files_external/3rdparty/icewind/streams/src/Wrapper.php line 132
    fclose(null)
 7. /var/www/virtual/XXX/html/cloud/lib/private/Files/Stream/HashWrapper.php line 75
    Icewind\Streams\Wrapper->stream_close()
 8. <<closure>>
    OC\Files\Stream\HashWrapper->stream_close()
 9. /var/www/virtual/XXX/html/cloud/apps/files_external/3rdparty/icewind/streams/src/Wrapper.php line 132
    fclose(null)
10. /var/www/virtual/XXX/html/cloud/lib/private/Files/Stream/HashWrapper.php line 75
    Icewind\Streams\Wrapper->stream_close()
11. /var/www/virtual/XXX/html/cloud/3rdparty/sabre/dav/lib/DAV/Server.php line 319
    OC\Files\Stream\HashWrapper->stream_close()
12. /var/www/virtual/XXX/html/cloud/apps/dav/appinfo/v1/publicwebdav.php line 113
    Sabre\DAV\Server->exec()
13. /var/www/virtual/XXX/html/cloud/public.php line 81
    require_once("/var/www/virtua ... p")

PUT /public.php/webdav/IMG_7587.JPG
from ***.***.****** at 2020-10-05T13:31:17+00:00
```

Screenshot after upload:

![Unbenannt](https://user-images.githubusercontent.com/58322182/95086797-31750580-0721-11eb-92a8-7dfbda020c66.PNG)

Screenshot after browser reload:

![Unbenannt2](https://user-images.githubusercontent.com/58322182/95086928-59fcff80-0721-11eb-850f-645f1cd1e5ca.PNG)

**Comment by bpgitkr** [comment]
Same issue. Upgraded to NextCloud 20 today from 19.0.3.

Steps to reproduce:

1. Create a new folder
2. Share the folder using a share link (doesn't matter whether you choose "File drop only" or "Allow uploads and editing")
3. Try to upload a file
4a. If you chose "Allow uploads and editing", the error will be "Unknown error occurred" and the file is only shows after you refresh the page.
4b. If you choose "File drop only", the error will be "Could not upload FILENAME", but the list of uploaded files shows the filename anyway. The file is successfully uploaded anyway.

**Comment by florian-obradovic** [comment]
Same here on two instances updatet from 19.0.4 to 20.0

**Comment by chrisingenhaag** [comment]
As a workaround you can try disabling the activities app. This works for me for my threema backup via webdav over shared link.

**Comment by florian-obradovic** [comment]
Confirmed. Disabling Activities App works for me.

**Comment by florian-obradovic** [comment]
Seems to be fixed and works with todays 20.0.1 update for me. Thx for your effort!

**Comment by devnull4242** [comment]
After update to 20.0.1 i activated the Activities App again and not it works fine. Thanks to the devs.

**Comment by florian-obradovic** [comment]
> After update to 20.0.1 i activated the Activities App again and not it works fine. Thanks to the devs.

You mean "now" not "not", right?

**Comment by devnull4242** [comment]
Yes. "now" and not "not". All works fine.

**Comment by Quiver92** [comment]
Hi! It's works when I turn completely turn off "Activities App". But how to solve this without turn it off?

**Comment by florian-obradovic** [comment]
It is fixed in 20.0.1

**Comment by szaimen** [comment]
based on florian-obradovic's comment I am closing this. Feel free to open a new issue if it should happen again.
